src/components/EditorSideBar/Blueprints.tsx

```diff
@@ -7,12 +7,20 @@ import { setActiveCourt } from "@/store/reducer/courtSpecDataSlice";
 import { mockTileData } from "../MockData/MockTileData";
 import { useStoreSelector } from "@/store/hooks";
 import { resetAll } from "@/store/reducer/canvasControlSlice";
+import { ICourtData } from "@/interfaces/design";
 
-const Blueprints: React.FC = () => {
+interface Props {
+  fetchedCourtsData: ICourtData[];
+}
+const Blueprints = ({ fetchedCourtsData }: Props) => {
   const dispatch = useDispatch();
   const [activateCourt, setActivateCourt] = useState<string>("");
   const { courtsData } = useStoreSelector((state) => state.courtSpecData);
 
+  const displayedCourt = fetchedCourtsData?.filter((item: ICourtData) => !item.isHidden);
```

Comment:
Do you think it filter should be on this component or redux slice? Or should I say is there any chance this hidden courts data will be used on any component in our app?

Reply:
I got your meaning. Actually, these hidden courts data will not be used on any components in CC-APP side, but these hidden courts data should be used on cc-admin to display the hidden data for the admins. It is unlike to filter these hidden courts data on the getAllCourtsData api at the cc-server side. But i think it is a good idea to filter those hidden courts data inside the useGetCourtsQuery()hook, using the selectFromResult from the hook's second parameter.

Reply:
try to use transformResponse
https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#customizing-query-responses-with-transformresponse

Reply:
Thanks guys, transformResponse is the better solution~